### ISSUE TITLE
Custom Validator for props should always run Before Computed properties

### ISSUE THREAD

**Issue by AllanOricil** [vuejs/vue#12286]
### What problem does this feature solve?
Vue 2.6.14 

Computed Properties are called before Props Custom Validators. 
I think It does not make sense, because If I declare a validator for my prop and it returns false, I think that Component shouldn't even be rendered.
I also think it shouldn't even waste computing time/resources calculating a computed prop dependant on a prop that its validator returned false. 
But Vue wastes time because the Validator is called after the computed prop method, meaning that the Computed Property does not know yet that the Validator returned false. Is there a reason for that?  

### What does the proposed API look like?
Custom Validator for Props should always run before Computed properties, when that computed property is dependant on the Prop that has a Custom Validator that returns false.
This way if the Prop validator returned false, Vue won't waste computing time/resources to calculate the Computed Property.

**Comment by posva** [comment]
This is probably expected depending on how you pass the prop. Most of the time validators will run before computed, but at the end it shouldn't matter. If you think this is a bug, open a new issue with a repro to provide more context

**Comment by AllanOricil** [comment]
@posva

"Most of the time validators will run before computed, but at the end it shouldn't matter"

It means you are aware of this already.
I really want to understand why the order does not matter. If the prop validation happens after the computed prop calculation, the computed prop will likely throw an exception, and to avoid it the developer has to add additional code to protect it, like use optional chainning.
I had to waste time to understand that this can happen. But it does not make any sense a computed prop to be calculated when its dependent of a prop that has a wrong value. 
It is counter intuitive. It could raise questions like "why should I also spend time adding protection on a computed prop if I have already added a Validator?"

Also, it is wasting time/resources calculating something where it should be calculate something else. Like, checking if other components are also receiving bad prop values, or rendering a template.

This could be something that would lead not only to performance improvement, but also DX improvement.
